src/components/modal/Modal.test.tsx

```diff
@@ -41,11 +40,33 @@ const renderWithModalRoot = (ui: JSX.Element, options: RenderOptions = {}) => {
   document.body.appendChild(appContainer)
   document.body.appendChild(modalContainer)
 
-  return render(ui, {
+  const renderResult = render(ui, {
     ...options,
     container: appContainer,
     baseElement: document.body,
   })
+
+  const modalWrapper = screen.getByRole('dialog')
+  const modalWindow = screen.getByTestId('modalWindow')
+
+  const queryForOpenButton = () =>
+    screen.queryByRole('button', {
+      name: 'Open default modal',
+    })
+  const queryForCloseButton = () =>
+    screen.queryByRole('button', {
+      name: 'Close this window',
+    })
+  const queryForOverlay = () => screen.queryByTestId('modalOverlay')
+
+  return {
+    ...renderResult,
+    modalWrapper,
+    modalWindow,
+    queryForOpenButton,
+    queryForCloseButton,
+    queryForOverlay,
+  }
```

Comment:
Rather than repeating these in many tests, just declare the helpers once and return them from the render method

One of my favorite RTL patterns (https://derekndavis.com/posts/maintainable-testing-react-testing-library)